### ISSUE TITLE
Generate a bug report URL that the user can just click to open an issue.

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#602]
> The best would be to generate a url that the user can just click to open an issue.
> See [this](https://github.com/sindresorhus/new-github-issue-url) project for example.

_Originally posted by @MarcoIeni in https://github.com/obi1kenobi/cargo-semver-checks/issues/311#issuecomment-1399352284_
            
This URL could be shown as part of `--bugreport` and could be used to automatically pre-fill a GitHub bug report with all the relevant data.

**Comment by dmatos2012** [comment]
Seeing that this looks like its still open, it looks like it would still be desirable. Looking at the structure of the issues that could be opened due to bugs(FP, FN, other) report, I was thinking that `cargo semver-checks --bugreport` could do one of two things:

1)
```
> cargo semver-checks --bugreport
Thanks for your report, what kind of bug is this
1. False Positive
2. False Negative
3. Other

> 2
```
Then this would open an issue on `False Negative`, and just populate what `--bugreport` would normally output on the body, and leave everything else for the person to fill. 

```
# feel free to actually go to this link so you know what I mean
Please follow the link https://github.com/obi1kenobi/cargo-semver-checks/issues/new?assignees=&labels=A-lint%2CC-enhancement&sys-info=output_of_bug_report&projects=&template=2-false-negative.yml for your report
```

The reason for the inputs would be to get the right `template`, but it would only fill the `--bugreport` text-area only in all of the cases. 

2)
```
# feel free to actually go to this link so you know what I mean
> cargo semver-checks --bugreport

Please follow the link https://github.com/obi1kenobi/cargo-semver-checks/issues/new?body=output_of_bug_report for your report.
```
No input needed, and just generate a blank issue with still the output of `--bugreport` as part of the body.

As far as the `impl` itself it doesnt look like we would need an specific `npm` tool for this? Because it would have to be installed etc, and I was thinking we could just hardcode the link, and just add the relevant `body` if 2) or `sys-info` if 1) to the link.

We could also decide to just open the browser for the person with a crate like [open](https://crates.io/crates/open), or let them open the browser themselves. 

Looking forward to your thoughts, and perhaps I can proceed to do an initial impl of this. @obi1kenobi

**Comment by obi1kenobi** [comment]
@dmatos2012 here are a few other issues that would be a good next step:
- Writing more lints is always welcome and a great way to contribute. It tends to have a lot of impact since it directly prevents SemVer breakage across the entire ecosystem.
- #863 would make our lives easier by speeding up our Windows CI builds.
- #462 is a complex issue, but the first part is easy and very helpful: make a test case that reproduces the problem in our test suite.
  - You'd make a new directory inside `test_crates/manifest_tests`, put whatever files are needed inside, and make a new snapshot test that runs `cargo-semver-checks` and records the outcome.
  - That outcome will initially be just a repro of the issue in #462. That way we'll know we can reproduce the problem and could analyze it better.
  - Then a future PR (either by you or someone else) can look into what's actually happening and why, and try to focus on fixing it. It'll be a much easier job because the problem is already reproduced, which is a solid head start.

These issues might seem tiny, but *they really aren't*. Projects like `cargo-semver-checks` most often die a "death by a thousand cuts", where instead of one major thing going wrong, it's instead a thousand little things all tripping you up wherever you turn. Fixing friction and minimizing maintenance burden is underrated, and extremely valuable. So everything from "faster CI" to "better test cases" is 💯 We can't fix things we can't test, and we certainly can't prevent them from breaking again either!
